compose.yml

```diff
@@ -41,117 +43,97 @@ services:
+      - ELASTIC_URL=http://elastic:9200
+      - ELASTIC_INDEX_NAME=huly_storage_index
+      - MONGO_URL=mongodb://mongodb:27017
+      - METRICS_CONSOLE=false
+      - METRICS_FILE=metrics.txt
+      - STORAGE_CONFIG=minio|minio?accessKey=minioadmin&secretKey=minioadmin
+      - REKONI_URL=http://rekoni:4004
+      - FRONT_URL=http://localhost:8087
+      - ACCOUNTS_URL=http://account:3000
+      - LAST_NAME_FIRST=${LAST_NAME_FIRST:-true}
+      - UPLOAD_URL=http${SECURE}://${HOST_ADDRESS}/files
+    restart: unless-stopped
+
+  collaborator:
+    image: hardcoreeng/collaborator:${HULY_VERSION}
+    environment:
+      - COLLABORATOR_PORT=3078
+      - SECRET=${SECRET}
+      - ACCOUNTS_URL=http://account:3000
+      - MONGO_URL=mongodb://mongodb:27017
+      - STORAGE_CONFIG=minio|minio?accessKey=minioadmin&secretKey=minioadmin
+    restart: unless-stopped
+
   account:
     image: hardcoreeng/account:${HULY_VERSION}
-    links:
-      - mongodb
-      - minio
-    ports:
-      - 3000:3000
     environment:
       - SERVER_PORT=3000
-      - SERVER_SECRET=${HULY_SECRET}
+      - SERVER_SECRET=${SECRET}
       - MONGO_URL=mongodb://mongodb:27017
-      - TRANSACTOR_URL=ws://transactor:3333;ws://${SERVER_ADDRESS}:3333
+      - TRANSACTOR_URL=ws://transactor:3333;ws${SECURE}://${HOST_ADDRESS}/_transactor
       - STORAGE_CONFIG=minio|minio?accessKey=minioadmin&secretKey=minioadmin
       - FRONT_URL=http://front:8080
       - MODEL_ENABLED=*
-      - ACCOUNTS_URL=http://${SERVER_ADDRESS}:3000
+      - ACCOUNTS_URL=http://localhost:3000
       - ACCOUNT_PORT=3000
     restart: unless-stopped
+
   workspace:
     image: hardcoreeng/workspace:${HULY_VERSION}
-    links:
-      - mongodb
-      - minio
     environment:
-      - SERVER_SECRET=${HULY_SECRET}
+      - SERVER_SECRET=${SECRET}
       - MONGO_URL=mongodb://mongodb:27017
-      - TRANSACTOR_URL=ws://transactor:3333;ws://${SERVER_ADDRESS}:3333
+      - TRANSACTOR_URL=ws://transactor:3333;ws${SECURE}://${HOST_ADDRESS}/_transactor
       - STORAGE_CONFIG=minio|minio?accessKey=minioadmin&secretKey=minioadmin
       - MODEL_ENABLED=*
       - ACCOUNTS_URL=http://account:3000
       - NOTIFY_INBOX_ONLY=true
     restart: unless-stopped
+
+  #love:
+  # image: hardcoreeng/love:${HULY_VERSION}
+  # restart: unless-stopped
   front:
     image: hardcoreeng/front:${HULY_VERSION}
-    links:
-      - mongodb
-      - minio
-      - elastic
-      - collaborator
-      - transactor
-    ports:
-      - 8087:8080
     environment:
       - SERVER_PORT=8080
-      - SERVER_SECRET=${HULY_SECRET}
-      - ACCOUNTS_URL=http://${SERVER_ADDRESS}:3000
-      - REKONI_URL=http://${SERVER_ADDRESS}:4004
-      - CALENDAR_URL=http://${SERVER_ADDRESS}:8095
-      - GMAIL_URL=http://${SERVER_ADDRESS}:8088
-      - TELEGRAM_URL=http://${SERVER_ADDRESS}:8086
+      - SERVER_SECRET=${SECRET}
+      - LOVE_ENDPOINT=http${SECURE}://${HOST_ADDRESS}/_love
+      - ACCOUNTS_URL=http${SECURE}://${HOST_ADDRESS}/_accounts
```

Comment:
Here and in other places you should add `${HTTP_PORT}` to the address, otherwise it won't work. I propose to include it into `HOST_ADDRESS` to simplify the configuration here.

Reply:
The goal is for huly to be served by a reverse proxy; No ports should be added.

Reply:
However if `HOST_ADDRESS` is an IP, the port is necessary.
I will look into that.

Reply:
You can easily test that the current configuration does not work, just configure some non-80 port and you won't be able to sign in.

Reply:
It does work, maybe your reverse proxy config is incorrect?

Reply:
It now also works with just IP, incase the `HOST_ADDRESS` is an IP we add the port to it so everything resolves correctly.

Reply:
```nginx
server {
    server_name huly.example.com;

    location / {
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;

      proxy_pass http://127.0.0.1:11040;
    }
  
    listen x.x.x.x:443 ssl; # managed by Certbot
    ...ssl
}
```
Heres my nginx config, just your basic config, everything works.